src/transformers/modeling_tf_utils.py

```diff
@@ -1152,6 +1153,68 @@ def __init__(self, config, *inputs, **kwargs):
     def get_config(self):
         return self.config.to_dict()
 
+    # TODO Matt: Rebase and replace all tf.keras with keras after merging the tf_keras PR.
+    #            Do not merge this before that PR is in!
+    @functools.wraps(tf.keras.Model.fit)
+    def fit(self, *args, **kwargs):
+        # Convert HF BatchEncoding objects to dicts that Keras understands
+        if args and isinstance(args[0], BatchEncoding):
```

Comment:
Does this work for multimodal models like vilt?

Reply:
OK, once the other PR is merged in we should run a few tests, just because I think there's cases when there might be e.g. a `BatchFeature` which contains tokenizer outputs. 

As a side note - why doesn't `BatchFeature` cause the same issue? Because it's not enforcing `int` type?

Reply:
Derp, you're totally right, it does! There was a bug in the code I was using to test that. Will update to convert `BatchFeature` as well, thanks for catching it!